staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package webhook implements the audit.Backend interface using HTTP webhooks.
+package webhook
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apimachinery/announced"
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/apis/audit/install"
+	"k8s.io/apiserver/pkg/apis/audit/v1alpha1"
+	"k8s.io/apiserver/pkg/util/webhook"
+)
+
+var (
+	groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+	groupVersions        = []schema.GroupVersion{v1alpha1.SchemeGroupVersion}
+	registry             = registered.NewOrDie("")
+
+	// Since the audit API isn't registered with the client or apimachinery API,
+	// create a unique scheme to register this API group with here.
+	//
+	// TODO(ericchiang): Refactor this once the audit API group is moved to k8s.io/api.
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	registry.RegisterVersions(groupVersions)
+	if err := registry.EnableVersions(groupVersions...); err != nil {
+		panic(fmt.Sprintf("failed to enable version %v", groupVersions))
+	}
+
+	install.Install(groupFactoryRegistry, registry, scheme)
+}
```

Comment:
This first bit, at least, is just copied from other webhooks.[1] The install call was because "k8s.io/apiserver/pkg/apis/audit/install" doesn't call this like other api groups.[2] Maybe we could have the audit/install package actually register the API group? 

[1] https://github.com/kubernetes/kubernetes/blob/e9b02c2e2b206ab0f4beb146e5fd1edcc385b70b/staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go#L210-L221
[2] https://github.com/kubernetes/kubernetes/blob/e9b02c2e2b206ab0f4beb146e5fd1edcc385b70b/pkg/apis/imagepolicy/install/install.go#L31-L33